tests/components/scene/test_init.py

```diff
@@ -1,149 +1,143 @@
+    reference to the original dictionary, instead of creating a copy, so
+    care needs to be taken to not modify the original.
+    """
+    light_1, light_2 = await setup_lights(hass, entities)
+    entity_state = {"state": "on", "brightness": 100}
+
+    assert await async_setup_component(
+        hass,
+        scene.DOMAIN,
+        {
+            "scene": [
+                {
+                    "name": "test",
+                    "entities": {
+                        light_1.entity_id: entity_state,
+                        light_2.entity_id: entity_state,
+                    },
+                }
+            ]
+        },
+    )
+    await hass.async_block_till_done()
+
+    await activate(hass, "scene.test")
+
+    assert light.is_on(hass, light_1.entity_id)
+    assert light.is_on(hass, light_2.entity_id)
+    assert 100 == light_1.last_call("turn_on")[1].get("brightness")
+    assert 100 == light_2.last_call("turn_on")[1].get("brightness")
+
+
+async def test_config_yaml_bool(hass, entities):
+    """Test parsing of booleans in yaml config."""
+    light_1, light_2 = await setup_lights(hass, entities)
+
+    config = (
+        "scene:\n"
+        "  - name: test\n"
+        "    entities:\n"
+        f"      {light_1.entity_id}: on\n"
+        f"      {light_2.entity_id}:\n"
+        "        state: on\n"
+        "        brightness: 100\n"
+    )
+
+    with io.StringIO(config) as file:
+        doc = yaml_loader.yaml.safe_load(file)
+
+    assert await async_setup_component(hass, scene.DOMAIN, doc)
+    await hass.async_block_till_done()
+
+    await activate(hass, "scene.test")
+
+    assert light.is_on(hass, light_1.entity_id)
+    assert light.is_on(hass, light_2.entity_id)
+    assert 100 == light_2.last_call("turn_on")[1].get("brightness")
+
+
+async def test_activate_scene(hass, entities):
+    """Test active scene."""
+    light_1, light_2 = await setup_lights(hass, entities)
+
+    assert await async_setup_component(
+        hass,
+        scene.DOMAIN,
+        {
+            "scene": [
+                {
+                    "name": "test",
+                    "entities": {
+                        light_1.entity_id: "on",
+                        light_2.entity_id: {"state": "on", "brightness": 100},
+                    },
+                }
+            ]
+        },
+    )
+    await hass.async_block_till_done()
+    await activate(hass, "scene.test")
+
+    assert light.is_on(hass, light_1.entity_id)
+    assert light.is_on(hass, light_2.entity_id)
+    assert light_2.last_call("turn_on")[1].get("brightness") == 100
+
+    calls = async_mock_service(hass, "light", "turn_on")
+
+    await hass.services.async_call(
+        scene.DOMAIN, "turn_on", {"transition": 42, "entity_id": "scene.test"}
+    )
+    await hass.async_block_till_done()
+
+    assert len(calls) == 1
+    assert calls[0].domain == "light"
+    assert calls[0].service == "turn_on"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)